src/wallet/mod.rs

```diff
@@ -1943,15 +1957,38 @@ pub(crate) mod test {
         let (wallet, _, _) = get_funded_wallet(get_test_wpkh());
         let checksum = wallet.descriptor_checksum(KeychainKind::External);
         assert_eq!(checksum.len(), 8);
+        assert_eq!(
+            get_checksum(&wallet.descriptor.to_string()).unwrap(),
+            checksum
+        );
+    }
 
-        let raw_descriptor = wallet
-            .descriptor
-            .to_string()
-            .split_once('#')
-            .unwrap()
-            .0
-            .to_string();
-        assert_eq!(get_checksum(&raw_descriptor).unwrap(), checksum);
+    #[test]
+    fn test_db_checksum() {
```

Comment:
Also wondering if its possible to reproduce the bug in test itself.. Master should fail in the test while this PR should pass. This backward compatibility test can stay too..

Reply:
We don't have the same logic for `get_checksum_bytes` in master though?

Reply:
Hmm.. On 2nd look it seems to me we are covering all the behaviors necessary.. I will think more if there's more needed to be covered..